avalon/tools/projectmanager/model.py

```diff
@@ -229,14 +229,18 @@ def set_assets(self, asset_ids):
             asset_tasks = asset.get("data", {}).get("tasks", [])
             tasks.update(asset_tasks)
 
-        # If no asset tasks are defined, use the project tasks.
-        if assets and not tasks:
-            project = io.find_one({"type": "project"})
-            tasks.update(
-                [task["name"] for task in project["config"].get("tasks", [])]
-            )
+        # # If no asset tasks are defined, use the project tasks.
```

Comment:
Why did you end up commenting this out? :)

Reply:
That's because I wasn't sure why Project manager showed project's tasks values if there are no tasks to show on selected asset and was scared to remove :) It had any reason?

Reply:
> That's because I wasn't sure why Project manager showed project's tasks values if there are no tasks to show on selected asset and was scared to remove :) It had any reason?

Ha, good - I like it your way too. ;) It's so confusing!

Anyhow, it got implemented here: https://github.com/getavalon/core/pull/330

I guess it was to match the behavior of the Launcher, which allowed to still go into an asset and work on a task when no explicit tasks where assigned. (Which I guess is backwards compatibility, since originally tasks were per project, not assigned per task.)

Reply:
I for one can't figure out what was the reasoning the in the first place. The only reason I can think of is the lack of project manager GUI so it was had to add individual tasks to assets, hence relying on a global task list. 

maybe @mottosso or @tokejepsen can shed some light on what is the use case?

Reply:
@mkolar sorry not sure what is the confusion?
Is it that the project manager shows the global tasks, or that you can see tasks per asset?

Reply:
What was the reason for project manager to show global tasks? Is the question. It seems that both us and colorbleed prefers only seeing tasks per asset, so I was wondering where the feature came from.

Reply:
Pretty sure that is a legacy feature from not having tasks per asset.

I would prefer having tasks per entity in Avalon, so you can explicitly say what tasks are available for the user.

Reply:
Great - my preference would also be then to remove them here too.

Reply:
In that case we'll remove the commented code completely as the resulting behavior is exactly as agreed here.